packages/services/api/src/modules/collection/providers/collection.provider.ts

```diff
@@ -0,0 +1,95 @@
+import { Injectable, Scope } from 'graphql-modules';
+import {
+  CreateDocumentCollectionInput,
+  CreateDocumentCollectionOperationInput,
+  UpdateDocumentCollectionInput,
+  UpdateDocumentCollectionOperationInput,
+} from '@/graphql';
+import { AuthManager } from '../../auth/providers/auth-manager';
+import { Logger } from '../../shared/providers/logger';
+import { Storage } from '../../shared/providers/storage';
+
+@Injectable({
+  global: true,
+  scope: Scope.Operation,
+})
+export class CollectionProvider {
+  private logger: Logger;
+
+  constructor(logger: Logger, private storage: Storage, private authManager: AuthManager) {
+    this.logger = logger.child({ source: 'CollectionProvider' });
+  }
+
+  getCollections(targetId: string) {
+    return this.storage.getPaginatedDocumentCollectionsForTarget({
+      targetId,
+      first: 100,
+      cursor: null,
+    });
+  }
+
+  getCollection(id: string) {
+    return this.storage.getDocumentCollection({ id });
+  }
+
+  getOperations(documentCollectionId: string) {
+    return this.storage.getPaginatedDocumentsForDocumentCollection({
+      documentCollectionId,
+      first: 100,
+      cursor: null,
+    });
+  }
+
+  getOperation(id: string) {
+    return this.storage.getDocumentCollectionDocument({ id });
+  }
+
+  async createCollection({ name, description, ...input }: CreateDocumentCollectionInput) {
+    const currentUser = await this.authManager.getCurrentUser();
+
+    return this.storage.createDocumentCollection({
+      createdByUserId: currentUser.id,
+      title: name,
+      description: description || '',
+      targetId: input.targetSelector.target,
+    });
+  }
+
+  deleteCollection(id: string) {
+    return this.storage.deleteDocumentCollection({ documentCollectionId: id });
+  }
+
+  async createOperation(input: CreateDocumentCollectionOperationInput) {
+    const currentUser = await this.authManager.getCurrentUser();
+    return this.storage.createDocumentCollectionDocument({
+      documentCollectionId: input.collectionId,
+      title: input.name,
+      contents: input.query,
+      variables: input.variables,
+      headers: input.headers,
```

Comment:
I think we should validate the input and set maximum lengths before inserting the data into the database? Errors can be exposed using the output object type ok error pattern

Reply:
(same applies for all other mutation fields added in this PR)